### PR TITLE
8335860: compiler/vectorization/TestFloat16VectorConvChain.java fails with non-standard AVX/SSE settings

### PR DESCRIPTION
**Pull request by jatin-bhateja**
Enabling test with explicit feature checks for x86 target.
Removing from test/hotspot/jtreg/ProblemList.txt

Best Regards,
Jatin

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8335860](https://bugs.openjdk.org/browse/JDK-8335860): compiler/vectorization/TestFloat16VectorConvChain.java fails with non-standard AVX/SSE settings (**Bug** - P4)

### Reviewers
 * [Sandhya Viswanathan](https://openjdk.org/census#sviswanathan) (@sviswa7 - **Reviewer**)
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20160/head:pull/20160` \
`$ git checkout pull/20160`

Update a local copy of the PR: \
`$ git checkout pull/20160` \
`$ git pull https://git.openjdk.org/jdk.git pull/20160/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20160`

View PR using the GUI difftool: \
`$ git pr show -t 20160`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20160.diff">https://git.openjdk.org/jdk/pull/20160.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20160#issuecomment-2226156586)

### CODE REVIEW

**Comment by sviswa7** [comment]
@jatin-bhateja  There was also a suggestion from @eme64 as part of https://github.com/openjdk/jdk/pull/20062 to remove @requires vm.compiler2.enabled from the test.

File: src/hotspot/cpu/x86/vm_version_x86.hpp [hunk 1]
```diff
@@ -834,7 +834,8 @@ class VM_Version : public Abstract_VM_Version {
     return true;
   }
 
-  // For AVX CPUs only. f16c support is disabled if UseAVX == 0.
+  // For AVX CPUs only since it needs VEX encoding which is missing on SSE targets,
+  // thus f16c support is disabled if UseAVX == 0.
```
Comment (sviswa7) [inline src/hotspot/cpu/x86/vm_version_x86.hpp:838]: This comment is somewhat or vey confusing. The code for supports_float16() by itself is very clear. I am wondering why do we need this explanation in the comment at all? Let us remove it altogether.

Comment (jatin-bhateja) [reply src/hotspot/cpu/x86/vm_version_x86.hpp]: Sure, FTR my comment was in relation to following check
https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/x86/vm_version_x86.cpp#L1056

 FP16-FP32 conversions are VEX encoded instructions and do not have SSE flavor. Thus, only works with UseAVX > 0.

**Comment by jatin-bhateja** [comment]
> @jatin-bhateja There was also a suggestion from @eme64 as part of #20062 to remove requires vm.compiler2.enabled from the test.

Test only validates specific C2 IR patten, framework makes sure to compile @Test annotated methods with top tier (c2 : default) compiler using Whitebox mechanism. So @require flags looks redundant here.  Agree.

**Comment by eme64** [comment]
Hi @jatin-bhateja 
Are you also going to address the questions for the VM changes from https://github.com/openjdk/jdk/pull/20062? That could be done in a separate RFE, but it would be nice to at least hear what you plan to do or not to do ;)

Or would you like to look into the VM changes @jaskarth ?

**Comment by jatin-bhateja** [comment]
> Hi @jatin-bhateja Are you also going to address the questions for the VM changes from #20062? That could be done in a separate RFE, but it would be nice to at least hear what you plan to do or not to do ;)
> 
> Or would you like to look into the VM changes @jaskarth ?

Hi @eme64 , Please go ahead and file RFE and kindly assign it to me. I will handle it as the part of RFE or close that with appropriate justifications. This PR is for test enablement.

**Review by sviswa7 — APPROVED** [review]
Looks good to me.

**Comment by vnkozlov** [comment]
Looks good. I submitted testing.

**Review by vnkozlov — APPROVED** [review]
My tier1-3 testing passed.

**Pull request closed without merging**